### PR TITLE
Feature/2.2.0 IKASAN-1646 rest client configuration

### PR DESCRIPTION
**Pull request by andrzej-majewski**

### CODE REVIEW

File: ikasaneip/configuration-service/src/main/java/org/ikasan/configurationService/metadata/JsonConfigurationMetaDataProvider.java [hunk 1]
```diff
@@ -69,7 +69,9 @@ public ConfigurationMetaData deserialiseMetadataConfiguration(String metaData) {
         try
         {
             //JSON file to Java object
-            result = this.mapper.readValue(metaData, List.class);
+            //result = this.mapper.readValue(metaData, List.class);
```
Comment (mitcje) [inline ikasaneip/configuration-service/src/main/java/org/ikasan/configurationService/metadata/JsonConfigurationMetaDataProvider.java]: Why leave this as commented - remove the line

File: ikasaneip/rest/rest-module-client/pom.xml [hunk 2]
```diff
@@ -129,6 +129,25 @@
             <scope>test</scope>
         </dependency>
 
+        <dependency>
+            <groupId>org.ikasan</groupId>
+            <artifactId>ikasan-configuration-service</artifactId>
+            <!--<scope>test</scope>-->
```
Comment (mitcje) [inline ikasaneip/rest/rest-module-client/pom.xml]: Remove commented out code?

File: ikasaneip/rest/rest-module-client/pom.xml [hunk 3]
```diff
@@ -129,6 +129,25 @@
...
+        </dependency>
+        <dependency>
+            <groupId>org.ikasan</groupId>
+            <artifactId>ikasan-standalone-persistence</artifactId>
+            <version>${project.version}</version>
+            <!--<scope>test</scope>-->
```
Comment (mitcje) [inline ikasaneip/rest/rest-module-client/pom.xml]: Remove commented out code?

File: ikasaneip/rest/rest-module-client/src/test/java/org/ikasan/rest/client/TestConfiguration.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
+package org.ikasan.rest.client;
```
Comment (mitcje) [inline ikasaneip/rest/rest-module-client/src/test/java/org/ikasan/rest/client/TestConfiguration.java]: What is this class actually doing?

File: ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/ConfigurationApplication.java [hunk 5]
```diff
@@ -255,4 +263,22 @@ public ResponseEntity getComponentsConfiguration(
         String configuredResources = configurationMetaDataExtractor.getComponentsConfiguration(flow);
         return new ResponseEntity(configuredResources, HttpStatus.OK);
     }
+
```
Comment (mitcje) [inline ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/ConfigurationApplication.java]: Removed commented out code ?

File: ikasaneip/configuration-service/src/main/java/org/ikasan/configurationService/metadata/JsonConfigurationMetaDataProvider.java (hunk 1 above)
Comment (andrzej-majewski) [reply ikasaneip/configuration-service/src/main/java/org/ikasan/configurationService/metadata/JsonConfigurationMetaDataProvider.java]: Removed

File: ikasaneip/rest/rest-module-client/pom.xml (hunk 2 above)
Comment (andrzej-majewski) [reply ikasaneip/rest/rest-module-client/pom.xml]: Removed

File: ikasaneip/rest/rest-module-client/pom.xml (hunk 3 above)
Comment (andrzej-majewski) [reply ikasaneip/rest/rest-module-client/pom.xml]: removed

File: ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/ConfigurationApplication.java (hunk 5 above)
Comment (andrzej-majewski) [reply ikasaneip/rest/rest-module/src/main/java/org/ikasan/rest/module/ConfigurationApplication.java]: removed

File: ikasaneip/rest/rest-module-client/src/test/java/org/ikasan/rest/client/TestConfiguration.java (hunk 4 above)
Comment (andrzej-majewski) [reply ikasaneip/rest/rest-module-client/src/test/java/org/ikasan/rest/client/TestConfiguration.java]: This Class is used by ConfigurationRestServiceImplTest in order to load configuration related beans to spring context. that is required to fully TEST Configuration REST endpoint

```
@ContextConfiguration(classes = { TestConfiguration.class }) 
```

**Pull request merged**